apps/theming-designer/src/components/ThemingDesigner.tsx

```diff
@@ -111,13 +112,38 @@ export class ThemingDesigner extends BaseComponent<{}, IThemingDesignerState> {
               <Samples backgroundColor={this.state.backgroundColor.str} textColor={this.state.textColor.str} />
             </ThemeProvider>
             <AccessibilityChecker theme={this.state.theme} themeRules={this.state.themeRules} />
-            <ThemeSlots theme={this.state.theme} themeRules={this.state.themeRules} />
+            <ThemeSlots
+              theme={this.state.theme}
+              themeRules={this.state.themeRules}
+              onFabricPaletteColorChange={this._onFabricPaletteColorChange}
+            />
           </Main>
         </Content>
       </Page>
     );
   }
 
+  private _onFabricPaletteColorChange = (newColor: IColor | undefined, fabricSlot: FabricSlots) => {
+    if (fabricPaletteColorChangeTimeout) {
+      clearTimeout(fabricPaletteColorChangeTimeout);
```

Comment:
You'll also want to invoke `clearTimeout` in `componentWillUnmount` or there's risk of a memory leak when the component is removed from the DOM.